src/impl_methods.rs

```diff
@@ -83,17 +83,43 @@ where
     }
 
     /// Return the shape of the array as it stored in the array.
+    ///
+    /// This is primarily useful for passing to other `ArrayBase`
+    /// functions, such as when creating another array of the same
+    /// shape and dimensionality.
+    ///
+    /// ```
+    /// use ndarray::Array;
+    ///
+    /// let a = Array::from_elem((2, 3), 5.);
+    ///
+    /// // Create an array of zeros that's the same shape and dimensionality as `a`.
+    /// let b = Array::<f64, _>::zeros(a.raw_dim());
+    /// ```
     pub fn raw_dim(&self) -> D {
         self.dim.clone()
     }
 
     /// Return the shape of the array as a slice.
```

Comment:
What would be really useful here is a doctest example. Something like
```rust
let a = array![
    [1, 2, 3, 4],
    [3, 4, 2, 1],
];
let shape = a.shape();
assert_eq!(shape, vec![2, 4].as_slice());
let b = ArrayBase::zeros(shape);
let c = Array::zeros(a.raw_dim());
assert_ne!(b, c);
assert_eq!(b, c.into_dyn());
```

Reply:
I've added an example based on what you provided. Is that better?

By the way, the reason why I didn't include `assert_ne!(b, c)` is that `PartialEq` is not implemented for two arrays of different dimensionality. Ideally, we'd support comparing `Array<A, D>` with `ArrayD<A>` for any `D: Dimension`, but that's not possible until Rust has specialization, because it would overlap the existing impl.

Reply:
It looks pretty neat to me - do we want to go ahead and merge this PR?

Reply:
Okay, I've merged it. Thanks for taking a look.